httptools/parser/parser.pyi

```diff
@@ -1,57 +1,59 @@
-from typing import Union, Any
+from typing import Union
 from array import array
 from .protocol import HTTPProtocol
 
 class HttpParser:
-    def __init__(self, protocol: Union[HTTPProtocol, Any]) -> None:
-        """
-        protocol -- a Python object with the following methods
-        (all optional):
-
-          - on_message_begin()
-          - on_url(url: bytes)
-          - on_header(name: bytes, value: bytes)
-          - on_headers_complete()
-          - on_body(body: bytes)
-          - on_message_complete()
-          - on_chunk_header()
-          - on_chunk_complete()
-          - on_status(status: bytes)
+    def __init__(self, protocol: HTTPProtocol | object) -> None:
```

Comment:
`object` is less permissive than `Any`.

---

Python still doesn't support "not required class method protocol": https://discuss.python.org/t/discussion-optional-class-and-protocol-fields-and-methods